### PR TITLE
Remove dashes from project UUID in snapshot names (#28)

### PR DESCRIPTION
**Pull request by hannes-ucsc**

### CODE REVIEW

**Review by aherbst-broad — APPROVED** [review]
Verified the regexes against our dev managed access dataset + snapshot names that drove this change, looks good.

**Comment by kbergin** [comment]
I'm going to ask Nikelle to review this in my place.

**Review by jessicaway — COMMENTED** [review]
Does this require any changes to the adapters?

**Comment by hannes-ucsc** [comment]
> I'm going to ask Nikelle to review this in my place.

@kbergin, could you try cancelling my the review request and requesting Nikelle directly here on the PR user interface?

> Does this require any changes to the adapters?

No, @jessicaway.

**Comment by kbergin** [comment]
Ok, fixed. @jessicaway 's review works instead too, just wanted it to be an eng from my team.

**Review by NoopDog — APPROVED** [review]

File: docs/dcp2_system_design.rst [hunk 1]
```diff
@@ -561,31 +561,51 @@ name. ::
...
+
+The ``project_id`` is the project UUID as allocated by the Ingest component. TDR
+does not allow dashes in dataset or snapshot names, so the ``project_id``
+field deviates from the standard string representation of a UUID by omitting the
+dashes.
+
+|nn| The standard representation can be reconstituted by inserting dashes as
+illustrated by the following Python code fragment::
+
+    '-'.join([x[0:8], x[8:12], x[12:16], x[16:20], x[20:36]])
```
Comment (ESapenaVentura) [inline docs/dcp2_system_design.rst]: I think there is a typo here, as UUIDs should only contain 32 chars. I would suggest either of these:

```suggestion
    '-'.join([x[0:8], x[8:12], x[12:16], x[16:20], x[20:]])
```
or
```suggestion
    '-'.join([x[0:8], x[8:12], x[12:16], x[16:20], x[20:32]])
```

**Review by ESapenaVentura — CHANGES_REQUESTED** [review]
Just a small comment on a typo, other than that LGTM 👍

**Review by jessicaway — APPROVED** [review]

**Review by nikellepetrillo — APPROVED** [review]

**Review by ruchim — APPROVED** [review]
looks good!

File: docs/dcp2_system_design.rst (hunk 1 above)
Comment (hannes-ucsc) [reply docs/dcp2_system_design.rst]: Fixed.

**Comment by hannes-ucsc** [comment]
Thanks for all the quick reviews. @ruchim and I discussed this today. It is possible that we won't need this since upcoming TDR changes eliminate the derivation of BQ dataset names from TDR snapshot/dataset names, removing the constraint that TDR snapshot or dataset names not contain dashes. If those changes land in the new prod instance of TDR soon enough, we will not need this.

We should sit on this for a bit until @ruchim knows when these TDR changes will land in the new prod instance of TDR. At that point we'll decide if we merge and later revert or if we just close this without merging. I want to avoid having mixed naming conventions for snapshots in TDR new prod, so we should decide before @aherbst-broad creates the first per-project snapshot in TDR new prod.

**Pull request merged**